packages/babel-helper-module-transforms/src/index.js

```diff
@@ -64,7 +66,11 @@ export function rewriteModuleStatementsAndPrepareHeader(
   }
 
   const headers = [];
-  if (hasExports(meta) && !strict) {
+  const defaultExportOnly = hasDefaultExportOnly(meta);
+  const doLegacyDefaultOnlyExport =
+    legacyDefaultOnlyExport && defaultExportOnly;
```

Comment:
Is it worth changing this to `legacyDefaultOnlyExport && hasDefaultExportOnly(meta)`, so we only run the check if we need to?

Reply:
Its updated, thanks! :)